test/mpi/runtests.in

```diff
@@ -769,9 +780,32 @@ sub RunMPIProgram {
     if ($test_opt->{init}) {
         $test_opt->{init}->();
     }
+
+    # acquire lock if requested
+    my $lockfile="/tmp/runtests-mem.lock";
+    my ($lock_sh, $lock_ex) = (1, 2);
+    my $got_lock;
+    if ($test_opt->{lock}) {
+        if ($test_opt->{lock} eq "shared") {
+            $got_lock = get_lock($lockfile, $lock_sh);
+        } else {
+            $got_lock = get_lock($lockfile, $lock_ex);
+        }
+    } elsif ($test_opt->{mem}) {
+        if ($test_opt->{mem} > $g_opt->{memory_total}) {
+            SkippedTest($programname, $np, $test_opt, $curdir, "xfail due to memory requirement");
+            next;
+        } elsif ($test_opt->{mem} * $g_opt->{memory_multiplier} > $g_opt->{memory_total} ) {
```

Comment:
I see memory multiplier initialized to one. How/when is it updated?

Reply:
It is set by environment variable `MPITEST_MEMORY_TOTAL` and `MPITEST_MEMORY_MULTIPLIER`, ref line 150.